interfaces/cython/cantera/reaction.pyx

```diff
@@ -302,6 +302,17 @@ cdef class Reaction:
         def __set__(self, allow):
             self.reaction.allow_negative_orders = allow
 
+    property input_data:
+        """
+        Get input data for this reaction with its current parameter values,
+        along with any user-specified data provided with its input (YAML)
+        definition.
+        """
+        def __get__(self):
+            cdef CxxAnyMap params
+            self.reaction.getParameters(params)
```

Comment:
Here and elsewhere: rather than calling into C++ twice, wouldn't it make sense to add an optional argument, i.e.
```C++
getParameters(AnyMap&  params, bool input=true)
```
Which eliminates the need for `mergeAnyMap`?

Reply:
The new `Reaction.parameters(bool withInput)` method has indeed made `mergeAnyMap` unnecessary.